src/item.cpp

```diff
@@ -2123,9 +2123,18 @@ void item::on_contents_changed()
     }
 }
 
-void item::on_damage( double, damage_type )
+double item::on_damage( double qty, damage_type )
```

Comment:
I'm against this change. Currently we don't have an event handler that returns non-void values (and that's a good thing: they should only react to changes). You can safely move this logic to mod_damage() (since on_damage() is invoked only from there).